src/Service/PasswordPathService.php

```diff
@@ -52,7 +57,7 @@ public function checkPasswordInPath(CategoryEntity $category, PageLoadedEvent|Ht
         }
     }
 
-    private function checkAuthenticated(PageLoadedEvent|HttpCacheHitEvent $event, string $navigationId)
+    private function checkAuthenticated(PageLoadedEvent|HttpCacheHitEvent|CoreHttpCacheHitEvent $event, string $navigationId)
```

Comment:
it seems enough to pass the request, not the $event to this method. Makes the typehint hopefully also less clunky.

Applies also for the methods that call this method.

Also you might want to consider to just pass the session and the request URI.